python/ray/data/datasource/datasource.py

```diff
@@ -149,9 +149,14 @@ def __init__(self, read_fn: Callable[[], Iterable[Block]], metadata: BlockMetada
         self._metadata = metadata
         self._read_fn = read_fn
 
-    def get_metadata(self) -> BlockMetadata:
+    @property
+    def metadata(self) -> BlockMetadata:
```

Comment:
Changing this from a method to a property for consistency with the newly-added `read_fn` property